lib/app/service/storage_service/browser_favicon_url_storage_service.dart

```diff
@@ -0,0 +1,59 @@
+import 'dart:async';
+
+import 'package:app/app/service/service.dart';
+import 'package:encrypted_storage/encrypted_storage.dart';
+import 'package:favicon/favicon.dart';
+import 'package:injectable/injectable.dart';
+import 'package:logging/logging.dart';
+
+const _browserFaviconURLDomain = 'browser_favicon_urls';
+
+final _suffixes = ['png', 'ico'];
+
+@singleton
+class BrowserFaviconURLStorageService extends AbstractStorageService {
+  BrowserFaviconURLStorageService(this._storage);
+
+  static final _log = Logger('BrowserFaviconURLStorageService');
+
+  /// Storage that is used to store data
+  final EncryptedStorage _storage;
+
+  Future<String?> getFaviconURL(Uri uri) async {
```

Comment:
Maybe we should store favicon for origin uri? Because uri may content other paths